### PR TITLE
Add advanced Options (subfolder, prefix, autostart)

### PR DESCRIPTION
**Pull request by mutschler**
This gives the ability to use the `folder` and/or `file_name_prefix`  features of metube based on the hostname of the video.

if either of the options is selected the hostname (with www. stripped) is appended to the request body for the given field.

EDIT: just saw #15 that might be something to include here as well...

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -13,25 +13,38 @@ chrome.runtime.onInstalled.addListener(function () {
...
         };
     }
 
+    let {hostname} = new URL(videoUrl)
+
+    let postData = {
+      "quality": "best",
+      "format": format,
+      "url": videoUrl,
+      'auto_start': !advancedSettings['disable_auto_start']
```
Comment (Rpsl) [inline src/background.js]: may you add check for default settings. it's need for case when user not saved their options after updating extension.

```javascript
'auto_start': !advancedSettings['disable_auto_start'] ?? true
```


screenshot-2024-03-22-22-38-5b27b937e2

**Comment by Rpsl** [comment]
thanks for you time! pls fix one little notice and i will publish new version in the store!

File: src/background.js (hunk 1 above)
Comment (mutschler) [reply src/background.js]: damn didn't think about that

**Comment by mutschler** [comment]
fixed :)

File: src/background.js [hunk 2]
```diff
@@ -13,25 +13,38 @@ chrome.runtime.onInstalled.addListener(function () {
...
+    let postData = {
+      "quality": "best",
+      "format": format,
+      "url": videoUrl,
+      'auto_start': !advancedSettings['disable_auto_start'] ?? true
+    }
+
+
+    Object.keys(advancedSettings).forEach((key) => {
+      if (advancedSettings[key] && !['disable_auto_start'].includes(key) ) {
```
Comment (Rpsl) [inline src/background.js:35]: it's need to be fixed here too.

Comment (mutschler) [reply src/background.js:35]: uhm... i thing the error is still for `'auto_start': !advancedSettings['disable_auto_start'] ?? true`

Comment (mutschler) [reply src/background.js:35]: maybe something like `'auto_start':  'disable_auto_start' in advancedSettings ? !advancedSettings['disable_auto_start'] : true`  will fix it?

Comment (Rpsl) [reply src/background.js:35]: Idk. I would try to make `advancedSettings` an object with properties and work with them. a little more code, but less abstraction

Comment (Rpsl) [reply src/background.js:35]: looked one more time. i think `(advancedSettings && !advancedSettings['disable_auto_start']) ?? true` will fix that

Comment (Rpsl) [reply src/background.js:35]: do you will fix that? or i will do it and publish

Comment (mutschler) [reply src/background.js:35]: Sorry i was busy and totally forgot about this. I've pushed another change (did end up going with the default value approach. You where right it's the cleaner solution)

**Comment by Rpsl** [comment]
thanks for you time. i published v1.7 to store

**Pull request merged**